src/Cms/File.php

```diff
@@ -451,9 +451,9 @@ public function parents()
 	 * Return the permanent URL to the file using its UUID
 	 * @since 3.8.0
 	 */
-	public function permalink(): string
+	public function permalink(): string|null
```

Comment:
Wouldn't it make sense to return the regular URL when UUIDs are disabled?

Reply:
Wasn't sure here. When somebody expects to get a permalink, is it better they fail when UUIDs are disabled (`null`) or better they receive something else, which is not a permalink though.

Reply:
I agree with Nico. Returning `null` is much safer. If the dev explicitly disables UUIDs, permalinks just won't be available.

Reply:
Ok, convinced.